### ISSUE TITLE
Notifier.ShowMessageDialog doesn't pick new line \n but accept <br>

### ISSUE THREAD

**Issue by aster94** [mit-cml/appinventor-sources#1856]
It would be nice to have this fixed, i noticed while using

reference: http://teach.appinventor.mit.edu/forums/topic/new-line-in-the-notifiers-showmessagedialog/

**Comment by ewpatton** [comment]
@mit-cml/cornell-cs-5152-students-fall-2019 One of you might want to take a look at this one.

**Comment by austenjoa** [comment]
I can take a look at this. Would there be any concerns of breaking already existing apps at all by changing the behavior of the message dialog?

**Comment by ewpatton** [comment]
@austenjoa Yes that is a possible concern if people are showing HTML that includes `<br>` and \n. There might be a couple ways to approach it:

1. We just replace \n with `<br>`. People will have to strip out the \n if they don't intend to have multiple line breaks.
2. We detect whether the string contains HTML. If it doesn't, then we interpret \n as `<br>` otherwise we keep the existing behavior.
3. We add a new html parameter to the method that defaults to true, with corresponding upgraders, so the developer can control how to interpret the contents of the string. The downside of this approach is that anyone using the companion to test must update due to the method signature change.
4. We do similar to 3 but make it a property, possibly controllable in the designer. Since default properties aren't sent, it will work the same in both old and new companion.

**Comment by austenjoa** [comment]
2 and 4 are probably the least intrusive options. I'm not sure which is preferable between them; 2 seems to be more user intuitive-- would there be any possibility of user disruption? Not sure if there's some use for newline characters that wouldn't always call for an html break instead. 4 seems safer, but I'm not sure how visible it would be to users (I'm users that think to look for html vs. normal interpretation of the string in the properties would likely already know to use html tags instead, anyways).
